boa_engine/src/builtins/object/mod.rs

```diff
@@ -833,25 +837,35 @@ impl Object {
         let builtin_tag = if o.is_array_abstract()? {
             utf16!("Array")
         } else {
-            // 6. Else if O has a [[ParameterMap]] internal slot, let builtinTag be "Arguments".
-            // 7. Else if O has a [[Call]] internal method, let builtinTag be "Function".
-            // 8. Else if O has an [[ErrorData]] internal slot, let builtinTag be "Error".
-            // 9. Else if O has a [[BooleanData]] internal slot, let builtinTag be "Boolean".
-            // 10. Else if O has a [[NumberData]] internal slot, let builtinTag be "Number".
-            // 11. Else if O has a [[StringData]] internal slot, let builtinTag be "String".
-            // 12. Else if O has a [[DateValue]] internal slot, let builtinTag be "Date".
-            // 13. Else if O has a [[RegExpMatcher]] internal slot, let builtinTag be "RegExp".
-            // 14. Else, let builtinTag be "Object".
-            match o.borrow().kind() {
-                ObjectKind::Arguments(_) => utf16!("Arguments"),
-                _ if o.is_callable() => utf16!("Function"),
-                ObjectKind::Error(_) => utf16!("Error"),
-                ObjectKind::Boolean(_) => utf16!("Boolean"),
-                ObjectKind::Number(_) => utf16!("Number"),
-                ObjectKind::String(_) => utf16!("String"),
-                ObjectKind::Date(_) => utf16!("Date"),
-                ObjectKind::RegExp(_) => utf16!("RegExp"),
-                _ => utf16!("Object"),
+            let o_borrow = o.borrow();
+
+            if o_borrow.is_arguments() {
+                // 6. Else if O has a [[ParameterMap]] internal slot, let builtinTag be "Arguments".
+                utf16!("Arguments")
+            } else if o.is_callable() {
+                // 7. Else if O has a [[Call]] internal method, let builtinTag be "Function".
+                utf16!("Function")
+            } else if o.is::<ErrorObject>() {
+                // 8. Else if O has an [[ErrorData]] internal slot, let builtinTag be "Error".
+                utf16!("Error")
+            } else if o.is::<bool>() {
```

Comment:
That was also my first thought, but on further thought it would be confusing for users to have an `Object<bool>` and an `Object<Boolean>` that have slightly different semantics.